### ISSUE TITLE
Tenant fails to be created when shareAddresses is used

### ISSUE THREAD

**Issue by ansmithf5** [F5Networks/f5-appsvcs-extension#279]
### Environment
 * Application Services Version: 3.20+
 * BIG-IP Version: All

### Summary
When multiple tenants in the same declaration are using the same shared virtual-address(es), some tenants fail to be created.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "class": "AS3",
    "persist": false,
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.20.0",
        "id": "Pool",
        "controls": {
            "class": "Controls",
            "trace": true,
            "logLevel": "debug",
            "traceResponse": true
        },
        "TEST_Shared1": {
            "class": "Tenant",
            "Application": {
                "class": "Application",
                "template": "generic",
                "test.item-foo": {
                    "class": "Service_Generic",
                    "virtualAddresses": [
                        "10.10.0.111",
                        "10.10.0.1"
                    ],
                    "virtualPort": 8080,
                    "shareAddresses": true
                }
            },
            "Application2": {
                "class": "Application",
                "template": "generic",
                "test.item-foo": {
                    "class": "Service_HTTP",
                    "virtualAddresses": [
                        "10.10.0.111",
                        "10.10.0.5"
                    ],
                    "virtualPort": 8079,
                    "shareAddresses": true
                }
            }
        },
        "TEST_Shared2": {
            "class": "Tenant",
            "Application": {
                "class": "Application",
                "template": "generic",
                "test.item-bar": {
                    "class": "Service_UDP",
                    "virtualAddresses": [
                        "10.10.0.111",
                        "10.10.0.222"
                    ],
                    "virtualPort": 8081,
                    "shareAddresses": true
                }
            }
        }
    }
}
```

2. Observe the following error response:
```json
"results": [
        {
            "code": 200,
            "message": "success",
            "lineCount": 30,
            "host": "localhost",
            "tenant": "TEST_Shared1",
            "runTime": 1329
        },
        {
            "code": 422,
            "message": "declaration failed",
            "response": "01020066:3: The requested Virtual Address (/Common/10.10.0.111) already exists in partition Common.",
            "host": "localhost",
            "tenant": "TEST_Shared2",
            "runTime": 1007
        }
    ]
```

### Expected Behavior
Both tenants should be successfully created.

### Actual Behavior
Only TEST_Shared1 is created while TEST_Shared2 is not.

**Comment by ansmithf5** [comment]
This is a known issue in AS3 3.20.0, and is being tracked internally in Jira with ticket AUTOTOOL-1592

**Comment by wncocz** [comment]
I'm just here to +1 this.  Trying to create a forwarding virtual on 0.0.0.0/0 (but a specific port), when there is already a forwarding virtual in /Common/ for anyport:

01020066:3: The requested Virtual Address (/Common/0.0.0.0) already exists in partition Common.

**Comment by dstokesf5** [comment]
As a workaround for this issue, you can submit the /Common/Shared declaration and your own tenant's declaration as two separate requests.

**Comment by wncocz** [comment]
In my case, the already-existing virtual-address /Common/0.0.0.0 is from a pre-AS3 implementation.  I POSTed my declaration, and I got that error.  So I edited the cli script __appsvcs_update, removed the "tmsh::create ltm virtual-address" line, and ran the script successfully.  It's not pretty, but I hope it will last until you release a fix.

**Comment by ansmithf5** [comment]
Good morning @wncocz, sorry for the delayed response. In order to reference a virtualAddress that's already on the box, and not in the declaration, you'll want to use the `bigip` key as in this example Service_Generic:

```json
"test.item-foo": {
    "class": "Service_Generic",
    "virtualAddresses": [
        {
            "bigip": "/Common/0.0.0.0"
        }
    ],
    "virtualPort": 8080,
    "shareAddresses": true
}
```

I just tested it out myself and it worked on a 13.1 box. Please let me know if you have any questions or difficulties.

**Comment by wncocz** [comment]
Thank you, @ansmithf5, that worked.

**Comment by ansmithf5** [comment]
Boom! Happy to help.

**Comment by wncocz** [comment]
I noticed this is not fixed in 3.21.0.  Are there plans to handle this consistently?

Our AS3 declarations are templated.  If we use { "bigip": "/Common/0.0.0.0" }, that works as long as the virtual-address has already been created.  If not, we must use just "0.0.0.0".  Then we must use { "bigip": "/Common/0.0.0.0" } in the next declaration.  This should just work using the "simple" format in all declarations, IMO.  We shouldn't need prior knowledge about the state of the BIGIP's virtual-addresses before submitting a declaration that uses shareAddresses.

**Comment by dstokesf5** [comment]
@wncocz This issue is fixed in 3.21.0. It sounds like the issue you are experiencing is a different one. This issue is specifically about one declaration and one POST request. The issue you are experiencing requires two requests, and I believe is a corner case with the address "0.0.0.0". I have created #319 based on my findings while looking into this.
